src/Avalonia.Themes.Fluent/TextBox.xaml

```diff
@@ -148,43 +135,134 @@
     <Setter Property="Foreground" Value="{DynamicResource TextControlForegroundFocused}" />
   </Style>
 
-  <Style Selector="TextBox:focus /template/ TextBlock#watermark, TextBox:focus /template/ TextBlock#floatingWatermark">
+  <Style Selector="TextBox:focus /template/ TextBlock#PART_Watermark, TextBox:focus /template/ TextBlock#PART_FloatingWatermark">
     <Setter Property="Foreground" Value="{DynamicResource TextControlPlaceholderForegroundFocused}" />
   </Style>
 
-  <Style Selector="TextBox:focus /template/ Border#border">
+  <Style Selector="TextBox:focus /template/ Border#PART_BorderElement">
     <Setter Property="Background" Value="{DynamicResource TextControlBackgroundFocused}"/>
     <Setter Property="BorderBrush" Value="{DynamicResource TextControlBorderBrushFocused}"/>
     <Setter Property="BorderThickness" Value="{DynamicResource TextControlBorderThemeThicknessFocused}" />
   </Style>
 
-  <Style Selector="TextBox:error /template/ Border#border">
+  <Style Selector="TextBox:error /template/ Border#PART_BorderElement">
     <Setter Property="BorderBrush" Value="{DynamicResource SystemControlErrorTextForegroundBrush}"/>
   </Style>
 
-  <Style Selector="TextBox /template/ DockPanel">
+  <Style Selector="TextBox /template/ DockPanel#PART_InnerDockPanel">
     <Setter Property="Cursor" Value="IBeam" />
   </Style>
+  
+  <Style Selector="TextBox /template/ TextBlock#PART_FloatingWatermark">
+    <Setter Property="IsVisible" Value="False" />
+  </Style>
+  <Style Selector="TextBox[UseFloatingWatermark=true]:not(TextBox:empty) /template/ TextBlock#PART_FloatingWatermark">
+    <Setter Property="IsVisible" Value="True" />
+  </Style>
 
-  <Style Selector="TextBox.revealPasswordButton">
+  <Style Selector="TextBox.revealPasswordButton[AcceptsReturn=False][IsReadOnly=False]:focus:not(TextBox:empty)">
     <Setter Property="InnerRightContent">
       <Template>
-        <ToggleButton IsChecked="{Binding $parent[TextBox].RevealPassword, Mode=TwoWay}">
-          <Panel Width="16" Height="16">
-            <DrawingPresenter Drawing="{DynamicResource EyeShowRegular}" IsVisible="{Binding $parent[TextBox].RevealPassword}"/>
-            <DrawingPresenter Drawing="{DynamicResource EyeHideRegular}" IsVisible="{Binding !$parent[TextBox].RevealPassword}"/>
-          </Panel>
-        </ToggleButton>
+        <ToggleButton Classes="PasswordBoxRevealButton"
+                      IsChecked="{Binding $parent[TextBox].RevealPassword, Mode=TwoWay}" />
       </Template>
     </Setter>
   </Style>
 
-  <Style Selector="TextBox.clearButton">
+  <Style Selector="TextBox.clearButton[AcceptsReturn=False][IsReadOnly=False]:focus:not(TextBox:empty)">
     <Setter Property="InnerRightContent">
       <Template>
-        <Button Content="Clear" Command="{Binding $parent[TextBox].Clear}" />
+        <Button Classes="TextBoxClearButton"
+                Command="{Binding $parent[TextBox].Clear}" />
```

Comment:
I would like to keep those buttons as simple as possible to reuse style TextBoxClearButton or PasswordBoxRevealButton if needed (move it to InnerLeftContent for example).